### PR TITLE
Fix #1033: skip two unit tests on Windows

### LINKED ISSUE #1033: Two unit tests fail on Windows
The following tests fail on Windows:
```
tests/test_openml/test_config.py::TestConfig::test_non_writable_home - AssertionError: 0 != 2
tests/test_utils/test_utils.py::OpenMLTaskTest::test__create_cache_directory - AssertionError: OpenMLCacheException not raised
```

### PR DESCRIPTION
**Pull request by mfeurer**

### CODE REVIEW

File: tests/test_runs/test_run_functions.py [hunk 1]
```diff
@@ -1618,7 +1618,9 @@ def test__run_task_get_arffcontent_2(self, parallel_mock):
             0.9655172413793104,
         ]
         scores = [v for k, v in res[2]["predictive_accuracy"][0].items()]
-        self.assertSequenceEqual(scores, expected_scores, seq_type=list)
+        np.testing.assert_array_almost_equal(
+            scores, expected_scores, decimal=2 if os.name == "nt" else 7
```
Comment (PGijsbers) [inline tests/test_runs/test_run_functions.py:1622]: I wasn't aware of this issue. This did not fail for me locally. Did you observe it in Appveyor?

Comment (mfeurer) [reply tests/test_runs/test_run_functions.py:1622]: Yes

Comment (mfeurer) [reply tests/test_runs/test_run_functions.py:1622]: Maybe that's due to the old python version of appveyor?

**Comment by mfeurer** [comment]
I'm surprised about the loky issue, I though we had fixed that in https://github.com/openml/openml-python/commit/f94672e672056806f3c3984f88b6d9d1c6cd368c#diff-062b8d0134bb5a02a7a6209393cf252284a8e0dd29483a36595cae93a981f725 by reordering the way the config is loaded...

File: tests/test_runs/test_run_functions.py (hunk 1 above)
Comment (PGijsbers) [reply tests/test_runs/test_run_functions.py:1622]: No clue, could be any number of version differences. But if it's accurate to two decimals it's should be fine nonetheless.

**Review by PGijsbers — APPROVED** [review]
Approving the PR since the changes make sense. The failing unit tests seem unrelated.

**Pull request merged**